core/src/apps/zcash/unified_addresses.py

```diff
@@ -0,0 +1,118 @@
+"""
+Implementation of encoding and decoding of Zcash
+unified addresses according to the ZIP-316.
+see: https://zips.z.cash/zip-0316
+"""
+
+from typing import Dict
+
+from trezor.crypto.bech32 import Encoding, bech32_decode, bech32_encode, convertbits
+from trezor.enums import ZcashReceiverTypecode as Typecode
+from trezor.utils import BufferReader, empty_bytearray
+from trezor.wire import DataError
+
+from apps.common.coininfo import CoinInfo
+from apps.common.readers import read_compact_size
+from apps.common.writers import write_bytes_fixed, write_compact_size
+
+from .f4jumble import f4jumble, f4unjumble
+
+
+def receiver_length(typecode: int):
+    """Byte length of a receiver."""
+    if typecode in [Typecode.P2PKH, Typecode.P2SH]:
+        return 20
+    if typecode in [Typecode.SAPLING, Typecode.ORCHARD]:
+        return 43
+    raise ValueError
+
+
+def prefix(coin: CoinInfo):
+    """Prefix for a unified address."""
+    if coin.coin_name == "Zcash":
+        return "u"
+    if coin.coin_name == "Zcash Testnet":
+        return "utest"
+    raise ValueError
+
+
+def padding(hrp: str) -> bytes:
+    assert len(hrp) <= 16
+    return hrp.encode() + bytes(16 - len(hrp))
+
+
+def encode(receivers: Dict[Typecode, bytes], coin: CoinInfo) -> str:
```

Comment:
please use py3.10-style annotations, i.e., in this case `dict[Typecode, bytes]` without importing `Dict`

Reply:
[fcc5fd1](https://github.com/trezor/trezor-firmware/pull/2398/commits/fcc5fd1cccbb92be5248f9179479e32e5856bb77)